backend/director/agents/video_generation.py

```diff
@@ -0,0 +1,179 @@
+                if not STABILITYAI_API_KEY:
+                    raise Exception("Stability AI API key not found")
+                video_gen_tool = StabilityAITool(api_key=STABILITYAI_API_KEY)
+                config_key = "stabilityai_config"
+            elif engine == "kling":
+                KLING_AI_ACCESS_API_KEY = os.getenv("KLING_AI_ACCESS_API_KEY")
+                KLING_AI_SECRET_API_KEY = os.getenv("KLING_AI_SECRET_API_KEY")
+                if not KLING_AI_ACCESS_API_KEY or not KLING_AI_SECRET_API_KEY:
+                    raise Exception("Kling AI API key not found")
+                video_gen_tool = KlingAITool(
+                    access_key=KLING_AI_ACCESS_API_KEY,
+                    secret_key=KLING_AI_SECRET_API_KEY,
+                )
+                config_key = "kling_config"
+
+            os.makedirs(DOWNLOADS_PATH, exist_ok=True)
+            output_file_name = f"video_{job_type}_{str(uuid.uuid4())}.mp4"
+            output_path = f"{DOWNLOADS_PATH}/{output_file_name}"
+
+            video_content = VideoContent(
+                agent_name=self.agent_name,
+                status=MsgStatus.progress,
+                status_message="Processing...",
+            )
+            self.output_message.content.append(video_content)
+
+            if job_type == "text_to_video":
+                prompt = text_to_video.get("prompt")
+                duration = text_to_video.get("duration", 5)
+                config = text_to_video.get(config_key, {})
+                if prompt is None:
+                    raise Exception("Prompt is required for video generation")
+                self.output_message.actions.append(
+                    f"Generating video for prompt: <i>{prompt}</i>"
+                )
+                self.output_message.push_update()
+                video_gen_tool.text_to_video(
+                    prompt=prompt,
+                    save_at=output_path,
+                    duration=duration,
+                    config=config,
+                )
+
+            self.output_message.actions.append(
+                f"Generated video saved at: <i>{output_path}</i>"
+            )
+            self.output_message.push_update()
+
+            # Upload to VideoDB
+            media = self.videodb_tool.upload(
+                output_path, source_type="file_path", media_type="video"
+            )
+            self.output_message.actions.append(
+                f"Uploaded generated video to VideoDB with Video ID: {media['id']}"
+            )
+            stream_url = media["stream_url"]
+            video_content.video = VideoData(stream_url=stream_url)
+            video_content.status = MsgStatus.success
+            video_content.status_message = "Video generated successfully"
+            self.output_message.push_update()
+            self.output_message.publish()
+
+        except Exception as e:
+            logger.exception(f"Error in {self.agent_name} agent: {e}")
```

Comment:
Set error status for Video_content

Reply:
in status_message ?

Reply:
imo, we should pass the error to Reasoning engine in form of AgentResponse.
And let reasoning engine figure out the the best way to convey the message ?

Reply:
for now, added the error in status message.
let me know if needs to be reverted

Reply:
reverted this  👍